src/core/common/permissions/validateRoleChange.ts

```diff
@@ -0,0 +1,68 @@
+import { isSiteModerationScoped } from "./isSiteModerationScoped";
+import {
+  isLTEModerator,
+  isModerator,
+  LTEModerator,
+  Moderator,
+  User,
+  UserRole,
+} from "./types";
+
+const validateModeratorRoleChange = (
+  viewer: Readonly<Moderator>,
+  user: Readonly<LTEModerator>,
+  role: UserRole,
+  scoped: boolean
+) => {
+  // Org mods may promote users to site moderators within their scope,
+  // but only if the user < org mod
+  const viewerIsOrgMod = !isSiteModerationScoped(viewer.moderationScopes);
+  const roleIsSiteMod = role === "MODERATOR" && scoped;
+  const userLTOrgMod =
+    isSiteModerationScoped(user.moderationScopes) || user.role !== "MODERATOR";
+
+  if (viewerIsOrgMod && userLTOrgMod && roleIsSiteMod) {
+    return true;
+  }
+
+  if (roleIsSiteMod) {
+    return true;
+  }
+
+  return false;
+};
+
+export const validateRoleChange = (
+  viewer: Readonly<User>,
+  user: Readonly<User>,
+  role: UserRole,
+  /**
+   * TODO: this optional argument that is only relevant
```

Comment:
I intend to leave this TODO here until we break MODERATOR into SITE_MODERATOR + ORG_MODERATOR